test/resource-providers/resource-proxy.test.ts

```diff
@@ -0,0 +1,118 @@
+        
+        expect(fn).toBeDefined();
+        expect(cloned.proxy).toBe(proxy);
+    });
+
+    test("When a stack is created with proxy for mastersRole in cluster provider, multiple stack can use it safely", () => {
+        const app = new App();
+        
+        const clusterProvider = new blueprints.GenericClusterProvider({
+            version: KubernetesVersion.V1_23,
+            mastersRole: blueprints.getResource(context => {
+                return Role.fromRoleName(context.scope, "mastersRole", "myrole");
+            }),
+        });
+
+        const builder = EksBlueprint.builder()
+            .clusterProvider(clusterProvider)
+            .account("123456789012")
+            .region("us-east-1");
+
+        const stack1 = builder.build(app, "resource-cluster");
+        const template = Template.fromStack(stack1);
+        
+        const stack2 = builder.build(new App(), "resource-cluster2");
+        const template1 = Template.fromStack(stack2);
+        // Then
+        expect(JSON.stringify(template.toJSON())).toContain(':iam::123456789012:role/myrole');
+        expect(nutil.isProxy(clusterProvider.props.mastersRole)).toBeTruthy();
+
+        
+        expect(JSON.stringify(template1.toJSON())).toContain(':iam::123456789012:role/myrole');
+        expect(nutil.isProxy(clusterProvider.props.mastersRole)).toBeTruthy();
+    });
+
+    test("When a stack is created with proxy for securityGroup in cluster provider, stack is created with proxy resolved", () => {
+        const app = new App();
+        
+        const sgDescription = "My new security group";
+        const clusterProvider = new blueprints.GenericClusterProvider({
+            version: KubernetesVersion.V1_23,
+            mastersRole: blueprints.getResource(context => {
+                return Role.fromRoleName(context.scope, "mastersRole", "myrole");
+            }),
+            securityGroup: blueprints.getResource(context => {
+                return new SecurityGroup(context.scope, 'ControlPlaneSG', {
+                    vpc: context.get(GlobalResources.Vpc)!,
+                    description: sgDescription
+                  });
+            })
+        });
+
+        const builder = EksBlueprint.builder()
+            .clusterProvider(clusterProvider)
+            .account("123456789012")
+            .region("us-east-1");
+
+        const stack1 = builder.build(app, "resource-cluster");
+        const template = Template.fromStack(stack1);
+        const templateString = JSON.stringify(template.toJSON());
+        expect(templateString).toContain(':iam::123456789012:role/myrole');
+        expect(templateString).toContain('ControlPlaneSG');
+        expect(templateString).toContain(sgDescription);
+        expect(nutil.isProxy(clusterProvider.props.securityGroup)).toBeTruthy();
+    });
+
+
+    // test("When a stack is created with proxy for launchTemplateSpec in cluster provider, stack is created with proxy resolved", () => {
```

Comment:
Could we remove this commented code if its not needed.

Reply:
addressed